### DISCUSSION TITLE
Add dark mode

### DISCUSSION THREAD

**Discussion by RakibRyan** [logisim-evolution/logisim-evolution#730] [Ideas]
white background is pretty hurting to eye. please add dark background.

**Comment by maehne** [comment]
Thanks for the suggestion. Actually, Logisim-evolution already provides two dark look and feels, which you can configure in the Preferences -> Window -> Look and Feel: "FlatLaf Dark" and "FlatLaf Darcula". In addition, you can configure the background color of the canvas in the same window. Is this close to what you are looking for? I am afraid that many icons are currently not well adapted to a dark theme though. See the related discussion on [replacing the built-in icons with SVG icons](https://github.com/logisim-evolution/logisim-evolution/discussions/640) for details. Help in this direction is welcome.

**Comment by MarcinOrlowski** [comment]
Just a note that LSe now allows user to configure colors of main design area (which I guess is the main source of pain at night).

**Reply by maehne** [reply]
We still have the issue that the component icons are not well adapted to dark mode, as you can observe from the following screenshot:

<img width="894" alt="grafik" src="https://user-images.githubusercontent.com/1174243/133474507-bcfddc17-c91b-4209-87fa-c6fae46e6084.png">

Black on dark gray gives simply not enough contrast.

**Reply by MarcinOrlowski** [reply]
I'd say that the UI is not really theme agnostic. Most icons are drawn in code that uses hard-coded colors etc.

**Comment by MarcinOrlowski** [comment]
Closing, as the dark theme(s) are here for some time.
